packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -0,0 +1,208 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/animation.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'scroll_metrics.dart';
+
+const double _kMinThumbExtent = 18.0;
+
+/// A [CustomPainter] for painting scrollbars.
+///
+/// Unlike [CustomPainter]s that sub-classes [CustomPainter] and only repaints
+/// when [shouldRepaint] returns true (which requires this [CustomPainter] to
+/// be rebuilt), this painter has the added optimization of repainting and not
+/// rebuilding when:
+///
+///  * the scroll position changes; and
+///  * when the scrollbar fades away.
+///
```

Comment:
These docs should talk about update() and how to use it

Reply:
Done